Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -889,6 +889,8 @@ void HTMLInputElement::attributeChanged(const QualifiedName& name, const AtomStr
         if (document().settings().switchControlEnabled()) {
             auto hasSwitchAttribute = !newValue.isNull();
             m_hasSwitchAttribute = hasSwitchAttribute;
+            if (attributeModificationReason != AttributeModificationReason::Directly)
+                return; // initializeInputTypeAfterParsingOrCloning has taken care of this.
```

Comment:
You mean initializeInputTypeAfterParsingOrCloning *will* take care of this.

Reply:
I copied this comment from the one we use for `typeAttr` and `valueAttr`, but maybe they're all wrong? The shadow tree creation will also happen even later when we resolve style it seems.